src/types.ts

```diff
@@ -292,3 +292,25 @@ export interface UploadResult {
   /** Root ID in the proof set */
   rootId?: number
 }
+
+/**
+ * Proofset data returned from the API
+ */
+export interface ProofsetData {
+  /** The proof set ID */
+  proofSetId: number
+  /** Array of root CIDs in the proof set */
+  roots: string[]
+  /** Additional metadata (structure may vary) */
+  metadata?: any
```

Comment:
this doesn't match the response we're dealing with.

See: https://github.com/filecoin-project/curio/blob/d359767f4beacbbba6ae90c86e5b4067d03da946/pdp/handlers.go#L492-L496

And also https://github.com/filecoin-project/curio/blob/2ea28495a0409c088835143ddddcb540a317925a/pdp/handlers.go#L521-L527

So your type here is going to need to be more structured.

If you want some test data to get this right, have a look at the proof set that I've been adding to here: `curl https://polynomial.computer/pdp/proof-sets/292 | jq`.

When building a validator for this data, you should validate the CIDs are actually CIDs. `isValidCommP` is in src/commp.ts that you can use for that.